### PR TITLE
Support enables allow cross-tenant replication via `allowCrossTenantReplication` in `Create` mode.

### PR DESCRIPTION
**Pull request by v-hongli1**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java [hunk 1]
```diff
@@ -219,14 +219,9 @@ public void canDisallowCrossTenantReplicationOnStorageAccount() {
...
                 .withNewResourceGroup(rgName)
                 .withSku(StorageAccountSkuType.STANDARD_LRS)
+                .allowCrossTenantReplication()
                 .create();
 
-        Assertions.assertFalse(storageAccount.isAllowCrossTenantReplication());
-
-        storageAccount.update()
-            .allowCrossTenantReplication()
-            .apply();
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java:228]: keep the update test, just change to disallow

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java:228]: The `allowCrossTenantReplication` interface has been added in `Create` mode, and this interface has been called in the test case `canDisallowCrossTenantReplicationOnStorageAccount`,
```
        StorageAccount storageAccount =
            storageManager
                .storageAccounts()
                .define(saName)
                .withRegion(Region.US_EAST2)
                .withNewResourceGroup(rgName)
                .withSku(StorageAccountSkuType.STANDARD_LRS)
                .allowCrossTenantReplication()
                .create();
```
 so 
```
        storageAccount.update()
            .allowCrossTenantReplication()
            .apply();
``` 
can be removed. However, verification of the initial setting value of `Cross-Tenant Replication` is still necessary.

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java:228]: Just add a 

```java
        storageAccount.update()
            .disallowCrossTenantReplication()
            .apply();
```

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java:228]: Removed the invoke of interface `disallowCrossTenantReplication` from the test case `canAllowCrossTenantReplicationOnStorageAccount`.

**Review by weidongxu-microsoft — APPROVED** [review]

**Review by XiaofeiCao — APPROVED** [review]

**Pull request merged**